src/cpp/session/modules/jobs/Job.cpp

```diff
@@ -208,7 +208,11 @@ json::Object Job::toJson() const
       error = r::sexp::getNames(actions_.get(), &names);
       if (!error)
       {
-         std::copy(names.begin(), names.end(), std::back_inserter(actions));
+         std::transform(
+            names.begin(),
+            names.end(),
+            std::back_inserter(actions),
+            [](const std::string& val) { return json::Value(val); });
```

Comment:
Curious why the new requirement for `json::Value` wrapping? It causes a lot of subtle pain points that I'd prefer us not to have to deal with, such as this one.

Reply:
I made all the conversion constructors explicit to avoid implicit conversions. There was actually an interesting issue that happened when I made the conversion constructors for Error (from boost error code) and Json::Value explicit. There was a method with three signatures:

`void method(Error, bool)`
`void method(boost::ec, json::Value)`
`void method(Error, json::Value, bool)`
 
The call looked like `method(boost::specific_ec, true);`. The compiler must have picked the correct method call to coerce to (the second one) originally, despite this being coercible to either the first or the second. I'm not sure how the compiler picked the right one.

I didn't know which it was supposed to be but had been fixing a lot of ec -> Error coercions, so I did that. I ended up getting infinite recursion because I was inside the 3rd overload, which called the 1st, which called the 3rd.

Reply:
Did we make the conversion operators for `json::Value` explicit just to get around this one call-site issue? IMO we should fix the calls you mention above so we don't force `json::Value(...)` calls all over the codebase.

Reply:
No, we made the conversion constructors explicit because implicit conversions can lead to unexpected behaviours in general. The example above just illuminated that.